File: src/boredBot/commands.ts
import { SlashCommandBuilder } from "discord.js";
import { Command } from "../discordBot/types";

export const boredBotCommands = [
	new SlashCommandBuilder()
		.setName(Command.Roll)
		.setDescription("Roll dice (e.g. 'd20', '6d12-4', '2d8 + 1d6+4')")
		.addStringOption((option) =>
			option
				.setName("input")
				.setDescription("The dice you want to roll")
				.setRequired(true),
		),
	new SlashCommandBuilder()
		.setName(Command.Help)
		.setDescription("Displays a list of available commands."),
];

File: DiscordBots/BoredBot/Commands.cs
using Discord;

namespace DiscordBots.BoredBot
{
    public static class BoredBotCommands
    {
        public static SlashCommandBuilder[] Commands => new[]
        {
            new SlashCommandBuilder()
                .WithName("roll")
                .WithDescription("Roll dice (e.g. 'd20', '6d12-4', '2d8 + 1d6+4')")
                .AddOption("input", ApplicationCommandOptionType.String, "The dice you want to roll", isRequired: true),
            
            new SlashCommandBuilder()
                .WithName("help")
                .WithDescription("Displays a list of available commands.")
        };
    }
}